internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -561,7 +561,7 @@ func (h *Host) modifyHostSettings(ctx context.Context, containerID string, req *
 					return err
 				}
 				defer func() {
-					if err != nil {
+					if retErr != nil {
```

Comment:
Does runExternalProcess below avoid this problem by not creating another err variable in the scope? If so which is the better pattern?

Reply:
yes, I've made a change to address this.